src/Serializer/YamlEncoder.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Serializer;
+
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+use Symfony\Component\Serializer\Encoder\YamlEncoder as BaseYamlEncoder;
+
+/**
+ * A YAML encoder with appropriate default options to embed the generated document into HTML.
+ */
+final class YamlEncoder implements EncoderInterface, DecoderInterface
+{
+    public function __construct(private readonly string $format, private readonly BaseYamlEncoder $yamlEncoder = new BaseYamlEncoder())
```

Comment:
```suggestion
    public function __construct(private readonly string $format, private readonly EncoderInterface $yamlEncoder = new BaseYamlEncoder())
```

This will fix the tests (in dev mode, encoders are decorated to be traceable).